src/main/java/com/cloudbees/plugins/credentials/CredentialsSelectHelper.java

```diff
@@ -596,24 +595,30 @@ public WrappedCredentialsStore(@NonNull ContextResolver resolver, @NonNull Crede
          * @throws ServletException if something goes wrong.
          */
         @RequirePOST
-        public void doAddCredentials(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException {
+        @Restricted(NoExternalUse.class)
+        public JSONObject doAddCredentials(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException {
             if (!store.isDomainsModifiable()) {
                 hudson.util.HttpResponses.status(400).generateResponse(req, rsp, null);
-                FormApply.applyResponse("window.alert('Domain is read-only')").generateResponse(req, rsp, null);
+                return new JSONObject()
+                        .accumulate("message", "Domain is read-only")
+                        .accumulate("notificationType", "ERROR");
```

Comment:
Any reason to use this over `#element`? We don't want the accumulate special behavior after all.

Reply:
> Any reason to use this over #element?

No particular reason, I wasn't sure which one was better to use.

Reply:
https://github.com/jenkinsci/credentials-plugin/pull/481/commits/d71602e7cc03a96bf876171da18a19d8dcb7efb9